### PR TITLE
Vim, but ready for merge now

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

Minature little mecha for critters (tiny simple animals) to ride around in. Comes with headlights, and the ability to chime or buzz (in case you cannot speak gal com)

Old PR #59736 

I got the sprites for it now!

## Why It's Good For The Game

Robotics content to explore and do, some pets can survive by hopping in these, they're pretty harmless fun

## Changelog
:cl:
expansion: Roboticists can now construct critter-sized mechas, start with an EVA helmet and a leg!
/:cl:

### CODE REVIEW

**Comment by MrStonedOne** [comment]
Hello, yes, thank you for your majestic pr sir and/or madam! If you could allow me this one question please and/or thank you!

How would a player, say, exit a vim?

**Review by Krysonism — APPROVED** [review]
Sprites are fucking approved, boi

File: code/modules/mob/living/simple_animal/bot/construction.dm [hunk 1]
```diff
@@ -526,3 +526,53 @@
...
+
+//Vim Assembly
+/obj/item/bot_assembly/vim
+	name = "incomplete vim assembly"
+	desc = "A space helmet with a leg attached to it. Looks like it needs another leg, if it is to become something."
+	icon_state = "vim_0"
+	created_name = "\improper Vim"
+
+/obj/item/bot_assembly/vim/attackby(obj/item/part, mob/user, params)
+	..()
```
Comment (Rohesie) [inline code/modules/mob/living/simple_animal/bot/construction.dm:538]: Any reason to neither preserve nor respect the parent call return?

File: code/modules/vehicles/cars/vim.dm [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+/obj/vehicle/sealed/car/vim/examine(mob/user)
+	. = ..()
+	. += span_notice("[src] can be repaired with a welder.")
+
+/obj/vehicle/sealed/car/vim/obj_destruction(damage_flag)
+	new /obj/effect/decal/cleanable/oil(get_turf(src))
+	do_sparks(5, TRUE, src)
+	visible_message(span_boldannounce("[src] blows apart!"))
+	. = ..()
```
Comment (Rohesie) [inline code/modules/vehicles/cars/vim.dm:33]:
```suggestion
	return ..()
```

File: code/modules/vehicles/cars/vim.dm [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+/obj/vehicle/sealed/car/vim/mob_try_enter(mob/entering)
+	if(!isanimal(entering))
+		return FALSE
+	var/mob/living/simple_animal/animal = entering
+	if(animal.mob_size != MOB_SIZE_TINY)
+		return FALSE
+	. = ..()
```
Comment (Rohesie) [inline code/modules/vehicles/cars/vim.dm:41]:
```suggestion
	return ..()
```

File: code/modules/clothing/spacesuits/miscellaneous.dm [hunk 4]
```diff
@@ -282,6 +282,22 @@ Contains:
 	flash_protect = FLASH_PROTECTION_NONE
 	armor = list(MELEE = 0, BULLET = 0, LASER = 0,ENERGY = 0, BOMB = 0, BIO = 100, RAD = 20, FIRE = 50, ACID = 65)
 
+/obj/item/clothing/head/helmet/space/eva/examine(mob/user)
+	. = ..()
+	. += span_notice("You can start constructing a critter sized mecha with a [span_bold("cyborg leg")].")
+
+/obj/item/clothing/head/helmet/space/eva/attackby(obj/item/attacked_with, mob/user, params)
+	. = ..()
+	if(!istype(attacked_with, /obj/item/bodypart/l_leg/robot) && !istype(attacked_with, /obj/item/bodypart/r_leg/robot))
```
Comment (Rohesie) [inline code/modules/clothing/spacesuits/miscellaneous.dm:291]: Respect the parent call returns. Return early if the parent call says to.

**Comment by TheBonded** [comment]
can you slot pAIs into it?

File: code/modules/vehicles/cars/vim.dm [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+	initialize_controller_action_type(/datum/action/vehicle/sealed/noise/buzz, VEHICLE_CONTROL_DRIVE)
+	initialize_controller_action_type(/datum/action/vehicle/sealed/headlights/vim, VEHICLE_CONTROL_DRIVE)
+
+/obj/vehicle/sealed/car/vim/update_overlays()
+	. = ..()
+	var/static/piloted_overlay
+	var/static/headlights_overlay
+	if(isnull(piloted_overlay))
+		piloted_overlay = iconstate2appearance(icon, "vim_piloted")
+		headlights_overlay = iconstate2appearance(icon, "vim_headlights")
```
Comment (Rohesie) [inline code/modules/vehicles/cars/vim.dm:86]:
```suggestion
	var/static/piloted_overlay = iconstate2appearance(icon, "vim_piloted")
	var/static/headlights_overlay = iconstate2appearance(icon, "vim_headlights")
```
Static values in proc calls will be initialized a single time.

**Comment by Rohesie** [comment]
Ready for merge after this small change.

File: code/modules/vehicles/cars/vim.dm (hunk 5 above)
Comment (MrStonedOne) [reply code/modules/vehicles/cars/vim.dm:86]: Bad suggestion.

Static values in proc calls are initialized at world load time with a src of null.

Comment (MrStonedOne) [reply code/modules/vehicles/cars/vim.dm:86]: image

Comment (MrStonedOne) [reply code/modules/vehicles/cars/vim.dm:86]: https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#icons-are-for-image-manipulation-and-defining-an-objs-icon-var-appearances-are-for-everything-else

Comment (Rohesie) [reply code/modules/vehicles/cars/vim.dm:86]: I was not aware of this, very nice.

**Pull request merged**